projects/canopy/src/lib/message-with-brand-icon/message-with-brand-icon.component.ts

```diff
@@ -0,0 +1,19 @@
+import { Component, HostBinding, Input, ViewEncapsulation } from '@angular/core';
+
+import type { BrandIconName } from '../brand-icon';
+
+@Component({
+  selector: 'lg-message-with-brand-icon',
+  templateUrl: './message-with-brand-icon.component.html',
+  styleUrls: ['./message-with-brand-icon.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class LgMessageWithBrandIconComponent {
+  @Input() iconName: BrandIconName;
```

Comment:
In cases where we use icons, such as the buttons or quick action, the icon component itself is inserted via a named content projection slot. I wonder if we should do that here as well for consistency, rather than via an input. It also feels a bit more like the Canopy style, as we use content projection a lot. Happy to discuss though.

Reply:
Yes, that's a good point - the only issue is that by allowing the user to add the brand icon by themselves they have access to the `size` input which we want control over for this specific component.

Reply:
Thats a good point, it could make it more consistent, it might also provide you with the extra css specificity you need to override the `!important`.

We have set a precedent where components can modify internal icon styles, this breaks the rule about components not being aware of other components, but feels like an ok exception.
e.g.
```css
.lg-accordion .lg-icon { /* this is ok */
  margin-right: var(--space-md);
}

.lg-accordion .lg-alert { /* this is not ok */
  margin-right: var(--space-md);
}
```

Reply:
Yes, agreed and if we tackle #488 in the near future the css code would go away anyway. I'll make that change 👍